File: DSA/DivyanshKumarSingh_590012175/Day_18_Question_3.py
def SortColors(a):
    l,m,h=0,0,len(a)-1
    while m<=h:
        if a[m] == 0:
            a[l],a[m]=a[m],a[l]
            l+=1
            m+=1
        elif a[m]==1:
            m+=1
        else:
            a[m],a[h]=a[h],a[m]
            h-=1
    
a=list(map(int,input().split()))
SortColors(a)
print(*a)

File: DSA/DivyanshKumarSingh_590012175/Day_18_Question_3.c
#include <stdio.h>
#include <stdlib.h>

void SortColors(int *a, int n)
{
    int l=0,m=0,h=n-1,t;
    while (m<=h)
    {
        if(a[m]==0)
        {
            t=a[l];
            a[l]=a[m];
            a[m]=t;
            l++;m++;
        }
        else if(a[m]==1)
        {
            m++;
        }
        else
        {
            t=a[m];
            a[m]=a[h];
            a[h]=t;
            h--;
        }
    }
}

int main()
{
    int n;
    scanf("%d", &n);
    int *a = malloc(n * sizeof(int));
    for (int i = 0; i < n; i++)
        scanf("%d", &a[i]);
    SortColors(a, n);
    for (int i = 0; i < n; i++)
        printf("%d", a[i]);
    free(a);
    return 0;
}